File: 386-lexicographical-numbers/lexicalOrder.cpp
#include <iostream>
#include <unordered_map>
#include <string>
#include <vector>
#include <stack>
using namespace std;

class Solution {
public:
    vector<int> lexicalOrder(int n) {
        vector<int> ret;
        if (n < 1) return ret;
        ret.push_back(1);

        int cur = 1;
        while (ret.size() < n) {
            while (cur <= n / 10) {
                cur *= 10;
                ret.push_back(cur);
            }
            cur++;
            while (cur % 10 == 0) cur /= 10;
            if (cur <= n) ret.push_back(cur);
        }

        return ret;
    }
};

int main() {
    Solution s = Solution();
    s.lexicalOrder(19);
    return 0;
}

File: 386-lexicographical-numbers/lexicalOrder.py
#!/usr/bin/env python3
# -*- coding: utf-8 -*-

class Solution(object):
    def lexicalOrder(self, n):
        """
        :type n: int
        :rtype: List[int]
        """
        ret = [1]
        cur = 1
        while len(ret) < n:
            while cur <= n / 10:
                cur *= 10
                ret.append(cur)
            cur += 1
            while cur % 10 == 0:
                cur /= 10
            if cur <= n:
                ret.append(cur)
        return ret

s = Solution()
print(s.lexicalOrder(19))
